ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -274,41 +258,102 @@ azure_terminate(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session)
     return 0;
 }
 
+// Helper to return a list of object names for the given location.
+static int
+azure_object_list_helper(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *directory,
+  const char *search_prefix, char ***dirlistp, uint32_t *countp, bool list_single)
+{
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
+    std::vector<std::string> objects;
+    std::string complete_prefix;
+
+    *countp = 0;
+
+    if (directory != nullptr) {
+        complete_prefix += directory;
+        // Add a terminating '/' if one doesn't exist.
```

Comment:
I think we can improve using cpp notation here, can you investigate into it? @lting0804

Reply:
Yep, can we use a cpp find function here instead?

Reply:
this seems like overengineering if we need to if there's a trailing `/`

Reply:
Hmm I see your point, I'm happy with the way it is!